setup/base/base.go

```diff
@@ -435,18 +435,21 @@ func (b *BaseDendrite) SetupAndServeHTTP(
 				}
 			} else {
 				if externalHTTPAddr.IsUnixSocket() {
-					_ = os.Remove(externalHTTPAddr.Address)
+					err := os.RemoveAll(externalHTTPAddr.Address)
```

Comment:
Why did you change this to `RemoveAll`? The address should be pointing to a file, not a directory.

Reply:
If you're worried about panicing on a "not found error" then you need to check it via `errors.Is(err, fs.ErrNotExist)`

Reply:
Just a nice function to do a cleanup which gives you only errors you care about, I can change it if you want.
Do you not use removeall for files?

Reply:
No because if you point this at a directory it'll remove that and all files within it.

Reply:
fixed